test/modules/MetricsValidator/src/ValidateNumberOfMessagesSent.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace MetricsValidator
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.Agent.Diagnostics;
+
+    public class ValidateNumberOfMessagesSent : TestBase
+    {
+        readonly string endpoint = Guid.NewGuid().ToString();
+
+        public ValidateNumberOfMessagesSent(TestReporter testReporter, IMetricsScraper scraper, ModuleClient moduleClient)
+            : base(testReporter, scraper, moduleClient)
+        {
+        }
+
+        public override async Task Start(CancellationToken cancellationToken)
+        {
+            Console.WriteLine($"Starting {nameof(ValidateNumberOfMessagesSent)}");
+            await this.TestNumberSent(10, cancellationToken);
+            await this.TestNumberSent(100, cancellationToken);
+
+            await this.TestBatch(1, 1, cancellationToken);
+            await this.TestBatch(10, 1, cancellationToken);
+            await this.TestBatch(1, 10, cancellationToken);
+            await this.TestBatch(10, 10, cancellationToken);
+        }
+
+        async Task TestNumberSent(int n, CancellationToken cancellationToken)
+        {
+            int prevSent = await this.GetNumberOfMessagesSent(cancellationToken);
+            await this.SendMessages(n, cancellationToken);
+            int newSent = await this.GetNumberOfMessagesSent(cancellationToken);
+            this.testReporter.Assert($"Reports {n} messages sent", n, newSent - prevSent);
+        }
+
+        async Task TestBatch(int n, int m, CancellationToken cancellationToken)
+        {
+            int prevSent = await this.GetNumberOfMessagesSent(cancellationToken);
+            await this.SendMessageBatches(n, m, cancellationToken);
+            int newSent = await this.GetNumberOfMessagesSent(cancellationToken);
+            this.testReporter.Assert($"Reports {n * m} for {n} batches of {n}", n * m, newSent - prevSent);
+        }
+
+        async Task<int> GetNumberOfMessagesSent(CancellationToken cancellationToken)
+        {
+            Console.WriteLine("Getting number of messages sent");
+            var metrics = (await this.scraper.ScrapeEndpointsAsync(cancellationToken)).ToArray();
+            Metric metric = metrics.FirstOrDefault(m => m.Name == "edgehub_messages_received_total" && m.Tags.TryGetValue("route_output", out string output) && output == this.endpoint);
+
+            return (int?)metric?.Value ?? 0;
```

Comment:
Nit: I think returning `0` is too trivial of a case. Might was well do something very outrage if the metric is null i.e. MIN_INT.

Reply:
It's actually a meaningful value. When no messages have been sent, the metric won;t be scrapped. Metrics only show up after their first value. So number of messages sent will never report 0.